R/histogram.R

```diff
@@ -23,16 +23,53 @@ histogramUI <- function(id, ...) {
 #'
 #' @param session Shiny server session object
 #'
-#' @param data Data used to render bar plot. See \code{\link[graphics]{barplot}} for reference.
+#' @param data Reactive value containing a data table. This is global data.
+#'             Desired subtables can be retrieved using \code{chosenCategories}
+#'             and \code{chosenValues} parameters.
+#'
+#' @param chosenCategories List of categories names that were already chosen
+#'                         when slicing the data.
+#'
+#' @param chosenValues List of categories values that were already chosen
+#'                     when slicing the data.
+#'
+#' @param numberOfSimulationTimes How many simulation times occurred in simulation.
 #'
 #' @return None. Invoked for the side-effect of rendering bar plot.
 #'
 #' @export
-#' @importFrom graphics barplot
+#' @importFrom graphics barplot hist axis
 #' @importFrom shiny renderPlot
+#' @importFrom utils head
 #' @rdname histogram
-histogram <- function(input, output, session, data) {
-  output$histogram <- renderPlot(
-    barplot(data)
-  )
+histogram <- function(input, output, session, data, chosenCategories,
```

Comment:
Histogram got too specific because of such semantics. It should still be a general module and all the calculations as to breaks and labels should be part of largePatches module.